cornucopia/src/read_schema.rs

```diff
@@ -0,0 +1,107 @@
+use miette::NamedSource;
+use postgres::Client;
+
+use crate::utils::db_err;
+
+use self::error::Error;
+
+#[derive(Debug, Clone)]
+pub struct Schema {
+    pub path: String,
+    pub sql: String,
+}
+
+impl From<Schema> for NamedSource {
+    fn from(m: Schema) -> Self {
+        Self::new(m.path, m.sql)
+    }
+}
+
+fn read_schema(path: String) -> Result<Schema, Error> {
+    match std::fs::read_to_string(&path) {
+        Ok(sql) => Ok(Schema { path, sql }),
+        Err(err) => Err(Error::Io { path, err }),
+    }
+}
+
+fn execute_schema(client: &mut Client, schema: Schema) -> Result<(), Error> {
+    client.batch_execute(&schema.sql).map_err(|err| {
+        let msg = format!("{:#}", err);
+        if let Some((position, msg, help)) = db_err(&err) {
+            Error::Postgres {
+                msg,
+                help,
+                src: schema.into(),
+                err_span: Some((position as usize..position as usize).into()),
+            }
+        } else {
+            Error::Postgres {
+                msg,
+                help: None,
+                src: schema.into(),
+                err_span: None,
+            }
+        }
+    })
+}
+
+/// Reads schema files from arg paths, if the path is a file we read it, if the path is a directory we read all .sql files it contains.
+pub fn read_schemas(client: &mut Client, paths: Vec<String>) -> Result<(), Error> {
+    for path in paths {
+        let metadata = std::fs::metadata(&path).map_err(|err| Error::Io {
+            path: path.clone(),
+            err,
+        })?;
+        if metadata.is_file() {
+            execute_schema(client, read_schema(path)?)?
```

Comment:
This will try to execute *any* file, whether it is `.sql` or not. I don't think this is intended?

Reply:
Yes, this provide a way to force execution on a file without having to change names